engine/src/main/java/org/terasology/rendering/opengl/FrameBuffersManager.java

```diff
@@ -41,20 +41,35 @@
 import java.util.Date;
 import java.util.Map;
 
-import static org.lwjgl.opengl.EXTFramebufferObject.glDeleteFramebuffersEXT;
-import static org.lwjgl.opengl.EXTFramebufferObject.glDeleteRenderbuffersEXT;
-
 /**
- * The Default Rendering Process class.
+ * The FrameBuffersManager generates and maintains a number of Frame Buffer Objects (FBOs) used throughout the
+ * rendering engine.
  *
+ * In most instances Frame Buffers can be thought of as 2D arrays of pixels in GPU memory: shaders write to them or
+ * read from them. Some buffers are static and never change for the lifetime of the manager. Some buffers are dynamic:
+ * they get disposed and regenerated, i.e. in case the display resolution changes. Some buffers hold intermediate
+ * steps of the rendering process and the content of one buffer, "sceneFinal", is eventually sent to the display.
+ * <br/>
+ * At this stage no buffer can be added or deleted: the list of buffers and their characteristics is hardcoded.
+ * <br/>
+ * The existing set of public methods is primarily intended to allow communication between this manager and other parts
+ * of the rendering engine, most notably the PostProcessor and the GraphicState instances and the shaders system.
```

Comment:
I see some brief comments on some of them elsewhere in the class, that seems pretty good anyway :-) Adding some general Javadoc like this is then excellent and sure one day in the future maybe more when modules can use them. We either need a nice spot to put some doc that can be pulled into the modding api docs automatically or maybe some small intro article/page somewhere (risks going out of date tho)

I guess one could go overkill on the docs a little bit and make a method to create/dispose each FBO *just* so you have a place to put the javadoc ... but I don't know if it is useful at that level (or something modules could hook into anyway)